### PR TITLE
Add Makefile mockgen targets

### LINKED ISSUE #2090: Add automatic mockgen for `pkg/mock`
### Proposal

There are several mock interfaces generated from sources for testing purposes in `pkg/mock`. These interfaces were generated manually by running commands:
```bash
mockgen -destination=pkg/mock/mock_scaling/mock_interface.go -package=mock_scaling -source=pkg/scaling/scale_handler.go
mockgen -destination=pkg/mock/mock_scaler/mock_scaler.go -package=mock_scalers -source=pkg/scalers/scaler.go
mockgen -destination=pkg/mock/mock_scale/mock_interfaces.go -package=mock_scale -source=/Users/zroubali/go/pkg/mod/k8s.io/client-go@v0.22.1/scale/interfaces.go
mockgen -destination=pkg/mock/mock_client/mock_interfaces.go -package=mock_client -source=/Users/zroubali/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.6/pkg/client/interfaces.go
```

We should add a target to `Makefile` to regenerate this mock interfaces automatically when there is a change in the source files. `mockgen` installation should be handled by as well, similar way how `kustomize` or other tools are being installed in `Makefile`:
https://github.com/kedacore/keda/blob/00b2972aed7786aa4dcfc73b73a3f7afb367461f/Makefile#L203-L213

### Use-Case

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by mihaitodor**
_Provide a description of what has been changed_

### Checklist

- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))
- [x] Tests have been added - will be triggered each time `make build` is run
- [x] A PR is opened to update our Helm chart ([repo](https://github.com/kedacore/charts)) *(if applicable, ie. when deployment manifests are modified)* - N/A
- [x] A PR is opened to update the documentation on ([repo](https://github.com/kedacore/keda-docs)) *(if applicable)* - N/A
- [x] Changelog has been updated

Fixes #2090.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -123,8 +123,19 @@ clientset-generate: ## Generate client-go clientset, listers and informers.
 pkg/scalers/liiklus/LiiklusService.pb.go: hack/LiiklusService.proto
 	protoc -I hack/ hack/LiiklusService.proto --go_out=plugins=grpc:pkg/scalers/liiklus
 
+.PHONY: mockgen-gen
+mockgen-gen: mockgen pkg/mock/mock_scaling/mock_interface.go pkg/mock/mock_scaler/mock_scaler.go pkg/mock/mock_scale/mock_interfaces.go pkg/mock/mock_client/mock_interfaces.go pkg/scalers/liiklus/mocks/mock_liiklus.go
```
Comment (mihaitodor) [inline Makefile:127]: `mockgen-gen` is not a great name, but not sure how else to call it to maintain consistency.

File: CHANGELOG.md [hunk 2]
```diff
@@ -25,6 +25,7 @@
 - TODO ([#XXX](https://github.com/kedacore/keda/pull/XXX))
 - ScaledJob: introduce MultipleScalersCalculation ([#2016](https://github.com/kedacore/keda/pull/2016))
 - Add Graphite Scaler ([#1628](https://github.com/kedacore/keda/pull/2092))
+- Add Makefile mockgen targets ([#2090](https://github.com/kedacore/keda/issues/2090))
```
Comment (zroubalik) [inline CHANGELOG.md:28]: Minor nit, could you please move this to `### Other` section, b/c it is not user facing issue. Thanks!

File: Makefile (hunk 1 above)
Comment (zroubalik) [reply Makefile:127]: Agree, but I can think of any better. Let's keep it this way :)

**Review by zroubalik — APPROVED** [review]
LGTM, thanks a lot!

File: CHANGELOG.md (hunk 2 above)
Comment (mihaitodor) [reply CHANGELOG.md]: Oops, done. I missed that part :)

**Comment by mihaitodor** [comment]
@zroubalik Looks like I was a bit too eager to rebase on main. The last commit broke the build since `func getFromAuthOrMeta` was added to both `pkg/scalers/mysql_scaler.go` and `pkg/scalers/mongo_scaler.go`.

**Comment by zroubalik** [comment]
> @zroubalik Looks like I was a bit too eager to rebase on main. The last commit broke the build since `func getFromAuthOrMeta` was added to both `pkg/scalers/mysql_scaler.go` and `pkg/scalers/mongo_scaler.go`.

yeah, we are solving this in a separate issue. I'll ping you once it is done so you can do the final rebase, thanks!

**Comment by zroubalik** [comment]
@mihaitodor rebase please

**Pull request merged**